pkg/controller/servicebindingrequest/binder.go

```diff
@@ -237,26 +239,39 @@ func (b *binder) updateSpecContainers(
 	if containers, err = b.updateContainers(containers); err != nil {
 		return nil, err
 	}
-	if err = unstructured.SetNestedSlice(obj.Object, containers, containersPath...); err != nil {
+	if err = unstructured.SetNestedSlice(obj.Object, containers, b.getContainersPath()...); err != nil {
 		return nil, err
 	}
 	return obj, nil
 }
 
+func (b *binder) getContainersPath() []string {
+	return strings.Split(b.sbr.Spec.ApplicationSelector.BindingPath.PodSpecPath.Containers, ".")
+}
+
+func (b *binder) getVolumesPath() []string {
+	return strings.Split(b.sbr.Spec.ApplicationSelector.BindingPath.PodSpecPath.Volumes, ".")
```

Comment:
is there a chance that this is called on not initialized value?

Reply:
I don't understand the change required here.

Reply:
The [same comment](https://github.com/redhat-developer/service-binding-operator/pull/513/files#r459312216) applies here.

Reply:
Done. 8471a6520097095587857d584c3df8585112f11c

Reply:
Moving under the api makes it a public API. I think that's not really necessary.